network/gossip/libp2p/network.go

```diff
@@ -249,17 +250,23 @@ func (n *Network) genNetworkMessage(channelID string, event interface{}, targetI
 		emTargets = append(emTargets, tempID[:])
 	}
 
-	// get origin ID (inplace slicing n.me.NodeID()[:] doesn't work)
+	// get origin ID
 	selfID := n.me.NodeID()
 	originID := selfID[:]
 
+	// get message type
+	// get event type, remove the asterisk prefix if present, remove package names if present
+	eventFqdn := strings.Split(strings.TrimLeft(fmt.Sprintf("%T", event), "*"), ".")
```

Comment:
an error proof and less expensive way (than using reflect) of deriving the message type

Reply:
:( you are right, fmt.Sprintf has this code `p.fmt.fmtS(reflect.TypeOf(arg).String())`. I was hoping to avoid `reflect` but I guess I can't.